python/dgl/distributed/partition.py

```diff
@@ -666,6 +666,65 @@ def _set_trainer_ids(g, sim_g, node_parts):
             g.edges[c_etype].data["trainer_id"] = trainer_id
 
 
+def _update_node_edge_map(node_map_val, edge_map_val, g, num_parts):
+    # Update the node_map_val to be contiguous.
+    ntype_ids = {ntype: g.get_ntype_id(ntype) for ntype in g.ntypes}
+    ntype_ids_reverse = {v: k for k, v in ntype_ids.items()}
+    for part_id in range(num_parts):
+        for ntype_id in list(ntype_ids.values()):
+            ntype = ntype_ids_reverse[ntype_id]
+            start_id = node_map_val[ntype][part_id][0]
+            end_id = node_map_val[ntype][part_id][1]
+            if not (start_id == -1 and end_id == -1):
+                continue
+            prev_ntype_id = (
+                ntype_ids[ntype] - 1
+                if ntype_ids[ntype] > 0
+                else max(ntype_ids.values())
+            )
+            prev_ntype = ntype_ids_reverse[prev_ntype_id]
+            if ntype_ids[ntype] == 0:
+                if part_id == 0:
+                    node_map_val[ntype][part_id][0] = 0
+                else:
+                    node_map_val[ntype][part_id][0] = node_map_val[prev_ntype][
+                        part_id - 1
+                    ][1]
+            else:
+                node_map_val[ntype][part_id][0] = node_map_val[prev_ntype][
+                    part_id
+                ][1]
+            node_map_val[ntype][part_id][1] = node_map_val[ntype][part_id][0]
+    # Update the edge_map_val to be contiguous.
+    etype_ids = {etype: g.get_etype_id(etype) for etype in g.canonical_etypes}
+    etype_ids_reverse = {v: k for k, v in etype_ids.items()}
+    for part_id in range(num_parts):
+        for etype_id in list(etype_ids.values()):
+            etype = etype_ids_reverse[etype_id]
+            start_id = edge_map_val[etype][part_id][0]
+            end_id = edge_map_val[etype][part_id][1]
+            if not (start_id == -1 and end_id == -1):
+                continue
+            prev_etype_id = (
+                etype_ids[etype] - 1
+                if etype_ids[etype] > 0
+                else max(etype_ids.values())
+            )
+            prev_etype = etype_ids_reverse[prev_etype_id]
```

Comment:
Same here, we need to put some comments here.

Reply:
see details in the above example